src/main/groovy/life/qbic/samplestatus/reporter/services/RealLimsQueryService.groovy

```diff
@@ -0,0 +1,145 @@
+                IApplicationServerApi.class,
+                applicationServerUrl + IApplicationServerApi.SERVICE_URL, serverTimeout)
+        sessionToken = this.openBisApplicationServerApi.login(openbisUser, openbisPassword)
+    }
+
+    /**
+     * <p>This method is called by the Spring framework, after an instance of this class
+     * has been created.</p>
+     * <p>We just check, if the authentication against openBIS has been successful
+     * which is the case, when there is a session token present.</p>
+     */
+    @PostConstruct
+    void init() {
+        if (!sessionToken) {
+            throw new AuthenticationException("Authentication against LIMS service failed.")
+        }
+    }
+
+    /**
+     * {@InheritDocs}
+     */
+    @Override
+    List<SampleUpdate> getUpdatedSamples(Instant updatedSince) {
+        SampleSearchCriteria criteria = new SampleSearchCriteria()
+        // we make sure that the barcode is set, otherwise the sample is of no interest to us
+        criteria.withProperty("QBIC_BARCODE").thatContains("Q")
+        // only fetch latest samples
+        criteria.withModificationDate().thatIsLaterThanOrEqualTo(Date.from(updatedSince))
+ 
+        // we need to fetch properties, as the sample status (and barcode) is contained therein
+        SampleFetchOptions fetchOptions = new SampleFetchOptions()
+        fetchOptions.withProperties()
+
+        SearchResult<Sample> result = v3.searchSamples(sessionToken, criteria, fetchOptions)
+        List<SampleUpdate> sampleUpdates = result.getObjects().stream().map( sample -> createSampleUpdate(sample) ).collect( Collectors.toList() )
+
+        return sampleUpdates
+    }
+
+    public static SampleUpdate createSampleUpdate(Sample limsSample) {
+
+        Map<String,String> properties = limsSample.getProperties()
+        String sampleBarcode = properties.get("QBIC_BARCODE")
+
+        life.qbic.samplestatus.reporter.Sample sample = new life.qbic.samplestatus.reporter.Sample(sampleBarcode)
+
+        Date modificationDate = sample.getModificationDate()
+        String updatedStatus = mapSampleStatus(properties.get("SAMPLE_STATUS"))
+
+        return new SampleUpdate(sample = sample, updatedStatus = updatedStatus, modificationDate = modificationDate)
+    }
+
+    /**
+     * This method does not make decisions based on which status should be updated. This decision might be made
+     * elsewhere, as the default status 'METADATA_REGISTERED´ should probably not overwrite later, lab-independent statuses.
+     */
+    private String mapSampleStatus(String statusString) {
+        switch (statusString) {
+            case "SAMPLE_RECEIVED":
+                return statusString
+            case "QC_PASSED":
+                return "SAMPLE_QC_PASS"
+            case "QC_FAILED":
+                return "SAMPLE_QC_FAIL"
+            case "LIBRARY_PREP_FINISHED":
+                return statusString
+            default:
+                return "METADATA_REGISTERED"
```

Comment:
I think this is dangerous. We should not assume the status `METADATA_REGISTERED` as default. Maybe you throw the exception here

Reply:
Like "Unkown sample status label". And if empty, you can write the exception message "Cannot map empty status label", sth like that maybe